src/adapter/mod.rs

```diff
@@ -234,11 +234,15 @@ fn encode<'a>(
             // number.
             let hyphens_before = if hyphens { group } else { 0 };
 
-            for idx in BYTE_POSITIONS[group]..BYTE_POSITIONS[group + 1] {
-                let b = bytes[idx];
+            for (idx, b) in bytes
```

Comment:
@kinggoesgaming Curious to know whether this change results in any regression in our microbenchmarks.

Reply:
hmm.. hadnt considered that.. I will have a look